driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/MongoClientImpl.java

```diff
@@ -78,6 +79,9 @@ private MongoClientImpl(final MongoClientSettings settings, final Cluster cluste
                             @Nullable final OperationExecutor executor,
                             @Nullable final Closeable externalResourceCloser) {
         this.settings = notNull("settings", settings);
+        if (settings.getTimeout(TimeUnit.MILLISECONDS) != null) {
```

Comment:
I'm not sure why you included this (esp as it's only in the reactive streams driver).  Can you explain the rationale?

Reply:
I added this as reactive streams driver work will come after all the sync work. So in theory they could be released in different releases. This explicitly tells users that its not supported in the rs driver.